extra/language/rust.nix

```
@@ -1,10 +1,26 @@
 { lib, config, pkgs, ... }:
 let
   cfg = config.language.rust;
+  strOrPackage = import ../../nix/strOrPackage.nix { inherit lib pkgs; };
+
+  hasLibraries = lib.length cfg.libraries > 0;
+  hasIncludes = lib.length cfg.includes > 0;
 in
 with lib;
 {
   options.language.rust = {
+    libraries = mkOption {
+      type = types.listOf strOrPackage;
+      default = [ ];
+      description = "Use this when another language dependens on a dynamic library";
```

Typo, but I still don't understand what this means.

```suggestion
      description = "Use this when another language depends on a dynamic library";
```

What would be another language? Can you make an example.

---

I think "another language" isn't really necessary (and a holdover from the equivalent C option), this should really be for any non-crate library dependencies (i.e. anything that would need pkg-config).